src/packageManager.ts

```diff
@@ -32,57 +33,45 @@ export class PackageManager extends EventEmitter {
         this.storage = storage;
         this.packages = {};
         this.version = version;
-        this.pckdir_watch = chokidar.watch(storage);
         this.ready = false;
 
-        this.pckdir_watch.on('addDir', async (pathDir) => {
-            let parsed = path.parse(pathDir);
-            if (parsed.dir === this.storage) {
-                logger.logInfo('Package added ' + parsed.name);
-                let http_p = await getport({
-                    port: getport.makeRange(52521, 52570),
-                });
-                let http_pp = await getport({
-                    port: getport.makeRange(52571, 52620),
-                });
-                this.packages[parsed.name] = new Package(
-                    `${this.storage}/${parsed.name}`,
-                    http_p,
-                    http_pp
-                );
-            }
+        this.pckdir_watch = chokidar.watch(storage, { depth: 0 });
+        this.pckdir_watch.on('addDir', async (path_dir) => {
+            let parsed = path.parse(path_dir);
+            logger.logInfo('Package added ' + parsed.name);
```

Comment:
Thanks for the improved fix with the watchers (#7)!

Unfortunately, this change dropped the

```js
            if (parsed.dir === this.storage) {
```

check, which results in the following error now:

```console
# camscripter-run 
CamScripter Launch
Package added packages
node:fs:585
  handleErrorFromBinding(ctx);
  ^

Error: ENOENT: no such file or directory, open '/usr/local/lib/node_modules/camscripter-raspberry/packages/packages/manifest.json'
    at Object.openSync (node:fs:585:3)
    at Object.readFileSync (node:fs:453:35)
    at Package.readManifest (/usr/local/lib/node_modules/camscripter-raspberry/dist/packageManager.js:189:31)
    at new Package (/usr/local/lib/node_modules/camscripter-raspberry/dist/packageManager.js:161:30)
    at PackageManager.<anonymous> (/usr/local/lib/node_modules/camscripter-raspberry/dist/packageManager.js:40:42)
    at Generator.next (<anonymous>)
    at fulfilled (/usr/local/lib/node_modules/camscripter-raspberry/dist/packageManager.js:5:58)
    at processTicksAndRejections (node:internal/process/task_queues:96:5) {
  errno: -2,
  syscall: 'open',
  code: 'ENOENT',
  path: '/usr/local/lib/node_modules/camscripter-raspberry/packages/packages/manifest.json'
}
```

Dropping the `if` looks cleaner to me as well. Adjusting the watcher seems to work for me too:

```js
this.pckdir_watch = chokidar.watch(`${storage}/*`, { depth: 0 });
```

#9 is my proposal to address this issue.